Learning Team Agenda - Aug 6th (#322)
emberjs/core-notes: learning-team/2020-08/2020-08-06.md

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [x] Chris Manson (CM)
- [x] Isaac Lee (IL)
- [x] Jared Galanis (JG)
- [x] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [x] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests

> - [] Your name here (INITIALS)

Reviewer: ```suggestion
- [x] Ava Wroten (AW)
- [] Your name here (INITIALS)
```

Me~

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [x] Chris Manson (CM)
- [x] Isaac Lee (IL)
- [x] Jared Galanis (JG)
- [x] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [x] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [x] Ava Wroten (AW)
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

### Mentorship programme

initiative update: 

- we ran a much smaller version of the mentorship programme for EmberConf
- Not a small amount of work to run it
  - guage interest
  - Chris and Locks are running a contributor workshop soon
  - refocus effort? 
- Should consider starting up the more perminantly remote version 

>   - what if there is no EmberConf next year :scream:

Reviewer 2: ```suggestion
  - what if there is no (in-person) EmberConf next year :scream:
```

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
<!-- ### Your topic (INITIALS, expected duration in minutes) -->

> <!-- replace with topic -->

Reviewer 2: ```suggestion
### Concretize making PRs for 2 Ember Guides issues (IL, 10-15 min)

IL - I think these feature requests will help in welcoming new visitors to the site. They do involve changing multiple files, possibly across multiple versions, so I'd like us to discuss how to divide the work involved so that each PR can be of reasonable size for review.

- [Replace "First" and "Last" name code examples](https://github.com/ember-learn/guides-source/issues/1480)
- [Fix API links in the Guides that go to "release"](https://github.com/ember-learn/guides-source/issues/1440)
```

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
<!-- ### Your topic (INITIALS, expected duration in minutes) -->
<!-- replace with topic -->

> <!-- replace with topic -->

Reviewer 2: ```suggestion
### Reorganizing`Built-In Components` in Ember Guides (IL, 10-15 min)

Related links:

  - https://github.com/ember-learn/guides-source/issues/1485
  - https://discuss.emberjs.com/t/ember-octane-and-2-way-binding-what-are-you-recommend/18092
  - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/input.hbs (used by `<Input>` component)
  - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/component.hbs (used by `<Textarea>` component)

It wasn't clear to the issuer that event handling for built-in `<Input>` component needs to be different for checkbox and non-checkbox inputs. Our checkbox examples show first and last names (not the best example of a checkbox).

Proposed todos:

- Rename the section from `Built-In Components` to `Input Components`
- Divide the section into 2 subsections: native inputs and built-in inputs
- The built-in inputs subsection has 3 sub-subsections: non-checkbox input, checkbox input, and textarea.
- Check that the event names are correct for the built-in non-checkbox and checkbox input components
```

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
<!-- ### Your topic (INITIALS, expected duration in minutes) -->

> ### Concretize making PRs for 2 Ember Guides issues (IL, 15-20 min)
> 
> IL - I think these feature requests will help in welcoming new visitors to the site. They do involve changing multiple files, possibly across multiple versions, so I'd like us to discuss how to divide the work involved so that each PR can be of reasonable size for review.
> 
> - [Replace "First" and "Last" name code examples](https://github.com/ember-learn/guides-source/issues/1480)
> - [Fix API links in the Guides that go to "release"](https://github.com/ember-learn/guides-source/issues/1440)

Reviewer 2: ```suggestion
### Concretize making PRs for 2 Ember Guides issues (IL, 15-20 min)

IL - I think these feature requests will help in welcoming new visitors to the site. They do involve changing multiple files, possibly across multiple versions, so I'd like us to discuss how to divide the work involved so that each PR can be of reasonable size for review.

- [Replace "First" and "Last" name code examples](https://github.com/ember-learn/guides-source/issues/1480)
- [Fix API links in the Guides that go to "release"](https://github.com/ember-learn/guides-source/issues/1440)
```
<!-- replace with topic -->

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
<!-- ### Your topic (INITIALS, expected duration in minutes) -->
<!-- replace with topic -->
### Reorganizing `Built-In Components` in Ember Guides (IL, 10-15 min)

Related links:

>   - https://github.com/ember-learn/guides-source/issues/1485
>   - https://discuss.emberjs.com/t/ember-octane-and-2-way-binding-what-are-you-recommend/18092
>   - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/input.hbs (used by `<Input>` component)
>   - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/component.hbs (used by `<Textarea>` component)

Reviewer 2: ```suggestion
- https://github.com/ember-learn/guides-source/issues/1485
- https://discuss.emberjs.com/t/ember-octane-and-2-way-binding-what-are-you-recommend/18092
- https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/input.hbs (used by `<Input>` component)
- https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/component.hbs (used by `<Textarea>` component)
```

It wasn't clear to the issuer that event handling for built-in `<Input>` component needs to be different for checkbox and non-checkbox inputs. Our checkbox examples show first and last names (not the best example of a checkbox).

Proposed todos:

- Rename the section from `Built-In Components` to `Input Components`

> - Divide the section into 2 subsections: native inputs and built-in inputs
> - The built-in inputs subsection has 3 sub-subsections: non-checkbox input, checkbox input, and textarea.

Reviewer 2: ``````suggestion
- Reorganize the section as follows
  ```
  Input Components
  │
  ├── Native Inputs
  │
  └── Built-In Inputs
      │
      ├── Non-Checkbox Input
      │
      ├── Checkbox Input
      │
      └── Textarea
  ```
- The built-in inputs subsection has 3 sub-subsections: non-checkbox input, checkbox input, and textarea.
``````

# Month Day, Year

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
<!-- ### Your topic (INITIALS, expected duration in minutes) -->
<!-- replace with topic -->
### Reorganizing `Built-In Components` in Ember Guides (IL, 10-15 min)

> Related links:
> 
> - https://github.com/ember-learn/guides-source/issues/1485
> - https://discuss.emberjs.com/t/ember-octane-and-2-way-binding-what-are-you-recommend/18092
> - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/input.hbs (used by `<Input>` component)
> - https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/component.hbs (used by `<Textarea>` component)
> 
> It wasn't clear to the issuer that event handling for built-in `<Input>` component needs to be different for checkbox and non-checkbox inputs. Our checkbox examples show first and last names (not the best example of a checkbox).
> 
> Proposed todos:
> 
> - Rename the section from `Built-In Components` to `Input Components`
> - Reorganize the section to:
>   ```
>   Input Components
>   │
>   ├── Native Inputs
>   │
>   └── Built-In Inputs
>       │
>       ├── Non-Checkbox Input
>       │
>       ├── Checkbox Input
>       │
>       └── Textarea
>   ```
> - Check that the event names are correct for the built-in non-checkbox and checkbox input components
> <!-- replace with topic -->
> <!-- replace with topic -->
> <!-- replace with topic -->

Reviewer 2: ```suggestion
#### Related links

- https://github.com/ember-learn/guides-source/issues/1485
- https://discuss.emberjs.com/t/ember-octane-and-2-way-binding-what-are-you-recommend/18092
- https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/input.hbs (used by `<Input>` component)
- https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/templates/component.hbs (used by `<Textarea>` component)

#### Problems

There are 3 problems that I'd like us to examine:

It wasn't clear to the issuer that event handling for built-in `<Input>` component needs to be different for checkbox and non-checkbox inputs (camel-cased versus dasherized). Since the underlying implementation for these inputs is different, we could have separate sub-subsections for checkbox and non-checkbox inputs.

Our checkbox examples show first and last names (not the best example of a binary/ternary attribute). They will need to be updated to help address https://github.com/ember-learn/guides-source/issues/1480.

I don't think we currently document how to use the `{{on}}` modifier for native inputs? It'd be good to have a subsection for native inputs for people who want to practice one-way data flow and who want to use `fillIn` test helper to test native inputs.

#### Proposed todos

- Rename the section from `Built-In Components` to `Input Components`
- Reorganize the section to:
  ```
  Input Components
  │
  ├── Native Inputs
  │
  └── Built-In Inputs
      │
      ├── Non-Checkbox Input
      │
      ├── Checkbox Input
      │
      └── Textarea
  ```
- Check that the event names are correct for the built-in non-checkbox and checkbox input components
<!-- replace with topic -->
<!-- replace with topic -->
<!-- replace with topic -->
```
